rust/src/common.rs

```diff
@@ -1,6 +1,23 @@
 use std::ffi::CString;
 use std::os::raw::c_char;
 
+pub mod nom7 {
+    use nom7::bytes::streaming::{tag, take_until};
+    use nom7::error::ParseError;
+    use nom7::IResult;
+
+    pub fn take_until_and_consume<'a, E: ParseError<&'a [u8]>>(t: &'a [u8])
+         -> impl Fn(&'a [u8]) -> IResult<&'a [u8], &'a [u8], E>
+    {
+        move |i: &'a [u8]| {
+            let t = t.clone();
```

Comment:
I don't think the clone is required here (hint from clippy).

Reply:
Indeed, this was a leftover from taking the nom combinator, and specializing it to `&[u8]`.
Fixed!